app/crud.py

```diff
@@ -0,0 +1,304 @@
+from sqlalchemy.orm import Session
+from . import models, schemas
+
+
+# -------------------------------------------------------------------------------------------------
+# User specific CRUD operations, # of functions = 5
+# -------------------------------------------------------------------------------------------------
+def get_user_by_id(db: Session, user_id: int) -> models.User:
+    """
+    This function queries the database for the User with the given user_id
+    and returns the User with the given user_id
+    :param db: The database session
+    :param user_id:
+    :return User:
+    """
+    return db.query(models.User).filter(models.User.user_id == user_id).first()
+
+
+def get_user_by_email(db: Session, email: str) -> models.User:
+    """
+    This function queries the database for the User with the given email
+    and returns the User with the given email
+    :param db: The database session
+    :param email:
+    :return User:
+    """
+    return db.query(models.User).filter(models.User.email == email).first()
+
+
+def get_users(db: Session, skip: int = 0, limit: int = 10) -> list[models.User]:
+    """
+    This function queries the database for Users with the given skip and limit boundaries
+    and returns a list of Users
+    :param db: The database session
+    :param skip: Starting index of the list, 0 by default
+    :param limit: Ending index (skip + limit), 10 by default
+    :return List[Type[models.User]]:
+    """
+    return db.query(models.User).offset(skip).limit(limit).all()
+
+
+def create_user(db: Session, user: schemas.UserCreate) -> models.User:
+    """
+    This function creates a new User
+
+    It uses a fake hashing 'algorithm', just appends "fakehashed" to the given password.
+
+    TODO: Change the hashing
+    :param db: The database session
+    :param user: schemas.UserCreate
+    :return: User
+    """
+    hashed_password = user.password + "fakehashed"
+
+    db_user = models.User(
+        email=user.email,
+        first_name=user.first_name,
+        last_name=user.last_name,
+        username=user.username,
+        age=user.age,
+        is_active=user.is_active,
+        user_type_id=user.user_type_id,
+        hashed_password=hashed_password
+    )
+    db.add(db_user)
+    db.commit()
+    db.refresh(db_user)
+    return db_user
+
+
+# def delete_user(db: Session, user_id: int) -> models.User:
```

Comment:
Whatever we've decided to go with hard or soft delete, the other option shouldn't stay commented there in the repo. In general, commented snippets of the code are not something we would prefer to commit on dev/main branches.